common/src/main/java/com/wynntils/core/chat/ChatModel.java

```diff
@@ -73,7 +76,7 @@ public final class ChatModel extends Model {
             Pattern.compile(" +§[47]Press §r§[cf](SNEAK|SHIFT) §r§[47]to continue§r$");
     private static final Pattern EMPTY_LINE_PATTERN = Pattern.compile("^\\s*(§r|À+)?\\s*$");
 
-    private static boolean extractDialog = false;
+    private static final Set<Feature> dialogExtractionDependents = new HashSet<>();
```

Comment:
Isn't there already a dependency system by `ManagerRegistry?` Why do we need a new one here?

Reply:
This one is an inner dependency of a manager (model)

Reply:
What do you mean by "inner dependency"?

Reply:
So ChatModel handles a few stuff related to chat, and npc dialog extraction is a optional thing it can do. This is one way to implement it but @magicus designed this, and I am not sure how it is meant to be used.

Reply:
Do you think we can extract this so we can use the ManagerRegistry dependency system? That'd be far nicer. Maybe we can split it into a ChatModel and NpcDialogueModel, and have the ChatManager pass "chat screens" to the latter, if it is enabled?

Reply:
(I'm not really fond of this `extractDialog` solution, there just was nothing better available when I wrote it)

Reply:
I tried it but the code was too mixed together. Perhaps you could do this better.

Reply:
Ok, I'll have a look at it later. Let's do it this way for now.